### ISSUE TITLE
Working with 8.11.1 ? 

### ISSUE THREAD

**Issue by ZacP** [nathanwoulfe/Preserver#6]
Hi,

We're trying to use this package on a project using Umbraco 8.11.1 and can't seem to get it to work.

Here's what we did:
1. Installed the package (1.2.0) via nuget
2. Added the property to the document type that we're testing (via a composition)
3. Tested the page but couldn't seem to get Preserver to work.  To test we went to a page in Umbraco, edited various fields (textstrings, media pickers, etc), didn't save, closed the tab, then opened up a new tab.  We tried this a few times with various levels of speed a few seconds to several minutes, but nothing happened.

Tested on Win 10, Edge latest version (90.0.818.51)

Thanks!

Zac

**Comment by nathanwoulfe** [comment]
Hi @ZacP, no idea of the top of my head, admittedly I haven't looked at this since I released it... Will try find some time to track down the issue

**Comment by ZacP** [comment]
Thanks, Nathan!  Just to double-check.  My test should have prompted preserver to ask to restore the content, correct?

**Comment by nathanwoulfe** [comment]
Yup, exactly that - it should be pushing the current editor state into localstorage, then offering to restore it when the page reloads.

**Comment by nathanwoulfe** [comment]
@ZacP I've had a bit of a poke around, found a few issues that would be breaking functionality in 8.11. 

Biggest issue is the BlockList editor. Most (all other) editors store a simple value - maybe a string, any array of UDI values, a JSON blob. BlockList has a complex model which I need to dig into and work out if I can update programmatically... If not, I'll still ship an updated version, with a caveat that it doesn't support Blocks.

**Comment by ZacP** [comment]
Thanks @***@***.***> a patch would be wonderful.  We aren’t using BlockList on this site so that won’t be an issue for us.

Thanks for all of your help

Zac

**Comment by nathanwoulfe** [comment]
@ZacP looks like it will work with BlockList, manage to wrangle it.

Not quite release-ready yet, but latest build are available here => https://ci.appveyor.com/project/nathanwoulfe/preserver/build/artifacts

**Comment by ZacP** [comment]
Hi Nathan,

Thanks for all of your work.  I tested out the new package and it still doesn't seem to be working.  I tested using the same steps as earlier.

Just to make sure, it's fine to add preserver to the doc type via a composition, correct?

**Comment by nathanwoulfe** [comment]
It should be, admittedly haven't tried it. Is there anything logged to the browser console?

**Comment by nathanwoulfe** [comment]
@ZacP new build should be up now - wasn't playing nice with invariant content. Works fine from a composition too. 

If this doesn't work, hopefully there's something in the console...

**Comment by ZacP** [comment]
Thanks, Nathan!  This is working now!  
I did find one other bug, let me know if I should create a new issue for it.

The media picker (Umbraco.MediaPicker out of the box property editor)  isn't restoring correctly as shown in the attachments.  There aren't any console errors.
This is on Windows 10, latest version of Edge.


![preserver-media-picker-bug-1](https://user-images.githubusercontent.com/5231911/117327624-0afe8080-ae50-11eb-87ec-cd768ce7be12.PNG)
![preserver-media-picker-bug-2](https://user-images.githubusercontent.com/5231911/117327626-0b971700-ae50-11eb-813b-976b794e943c.PNG)

**Comment by ZacP** [comment]
Hi Nathan,

This seems to be working great in our tests so far!  If anything changes, we’ll let you know.

Thanks for creating this great package

Zac

**Comment by nathanwoulfe** [comment]
Awesome, glad to hear. I'll hold on for a few days in case anything crops up, then will push out a proper release.
